optuna/samplers/_nsga2/crossover.py

```diff
@@ -0,0 +1,503 @@
+                param = (
+                    parents[0].params[param_name]
+                    if rng.rand() < swapping_prob
+                    else parents[-1].params[param_name]
+                )
+                child[param_name] = param
+
+        if len(numerical_distributions) == 0:
+            if _is_contained(child, search_space):
+                break
+            continue
+
+        # Processing from here on is applied only for numerical parameters.
+        parents_numerical_params_array = np.stack(
+            [
+                numerical_transform.transform(
+                    {
+                        param_key: parent.params[param_key]
+                        for param_key in numerical_search_space.keys()
+                    }
+                )
+                for parent in parents
+            ]
+        )  # Parent individual with NUMERICAL_DISTRIBUTIONS parameter.
+        if crossover_name == "uniform":
+            params_array = _uniform(
+                parents_numerical_params_array[0],
+                parents_numerical_params_array[1],
+                rng,
+                swapping_prob,
+            )
+        elif crossover_name == "blxalpha":
+            alpha = 0.5
+            params_array = _blxalpha(
+                parents_numerical_params_array[0], parents_numerical_params_array[1], rng, alpha
+            )
+        elif crossover_name == "sbx":
+            if len(study.directions) == 1:
+                eta = 2
+            else:
+                eta = 20
+            params_array = _sbx(
+                parents_numerical_params_array[0],
+                parents_numerical_params_array[1],
+                rng,
+                numerical_distributions,
+                eta,
+            )
+        elif crossover_name == "vsbx":
+            if len(study.directions) == 1:
+                eta = 2
+            else:
+                eta = 20
+            params_array = _vsbx(
+                parents_numerical_params_array[0], parents_numerical_params_array[1], rng, eta
+            )
+        elif crossover_name == "undx":
+            sigma_xi = 0.5
+            sigma_eta = 0.35 / np.sqrt(len(parents_numerical_params_array[0]))
+            params_array = _undx(
+                parents_numerical_params_array[0],
+                parents_numerical_params_array[1],
+                parents_numerical_params_array[2],
+                rng,
+                sigma_xi,
+                sigma_eta,
+            )
+        elif crossover_name == "undxm":
+            _m = len(parents_numerical_params_array) - 2
+            _n = len(parents_numerical_params_array[0])
+            if _n - _m <= 0:
+                raise RuntimeError(
+                    "undxm must be n > m-2,"
+                    " where m is the number of parental individuals to be selected"
+                    " at crossover (m=4) and n is the number of parameters of the individuals."
+                )
```

Comment:
Here, `m` is not `the number of parents`, but `the number of parents - 2`.
Therefore, to clarify the difference, we set `_m` (`_n` is the same as the usual the number of parameters).

However, since `n` and `m` are difficult to understand, we will change the names of these variables to `n_parents` and `n_params`.